File: p5js/js/sf6_classes.js
"use strict";

var sketch_6 = function(p) {

	var numDots = 10;
	var dots = [];

	p.setup = function() {
		p.createCanvas(640, 360);

		for (var i=0; i<numDots; i++) {
    		dots[i] = new Dot(100, 100);
  		}
	}

	p.draw = function() {
		p.background(127);

		for (var i=0; i<dots.length; i++) {
			dots[i].run();
		}
	}

	// ~ ~ ~

	class Dot {

		constructor(_x, _y) {
			this.pos = p.createVector(_x, _y);
			this.s = 50;
			this.speed = p.random(1, 5);
			this.target = p.createVector(p.width, p.random(p.height));

			this.fillOrig = p.color(0, 100, 255);
			this.fillHit = p.color(0, 200, 255);
			this.fillNow = this.fillOrig;
		}

		update() {    
    		this.pos.x = p.lerp(this.pos.x, this.target.x, 0.005 * this.speed);
    		this.pos.y = p.lerp(this.pos.y, this.target.y, 0.005 * this.speed);
	
		    if (this.target.x > 0 && this.pos.x > p.width - (this.s/2) || this.target.x < 0 && this.pos.x < -(this.s/2)) {
		    	this.target.x *= -1;
		    	this.target.y = p.random(p.height);
		    }

			if (this.target.x < 0) {
				this.fillNow = this.fillHit;
			} else {
				this.fillNow = this.fillOrig;
			}
		}

		draw() {
			p.fill(this.fillNow);
			p.ellipse(this.pos.x, this.pos.y, this.s, this.s);    
		}

		run() {
			this.update();
			this.draw();
		}

	}

}

var p5_6 = new p5(sketch_6, "canvas6");

File: p5py/sf6_classes.py
from p5 import *

numDots = 10
dots = []

def setup():
    global numDots, dots
    size(640, 360)
    
    for i in range(0, numDots):
        d = Dot(100, 100)
        dots.append(d)

def draw():
    global numDots, dots
    background(127)
    
    for d in dots:
        d.run()

# ~ ~ ~ ~

class Dot:

    def __init__(self, _x, _y):
        self.pos = Vector(_x, _y)
        self.s = 50
        self.speed = random_uniform(1, 5)
        self.target = Vector(width, random_uniform(height))
        self.fillOrig = Color(0, 100, 255)
        self.fillHit = Color(0, 200, 255)
        self.fillNow = self.fillOrig
        
    def update(self):        
        self.pos.x = lerp(self.pos.x, self.target.x, 0.005 * self.speed)
        self.pos.y = lerp(self.pos.y, self.target.y, 0.005 * self.speed)
        
        if (self.target.x > 0 and self.pos.x > width - (self.s/2)) or (self.target.x < 0 and self.pos.x < -(self.s/2)):
            self.target.x *= -1
            self.target.y = random_uniform(height)
            
        if self.target.x < 0:
            self.fillNow = self.fillHit
        else:
            self.fillNow = self.fillOrig
    
    def draw(self):
        fill(self.fillNow)
        ellipse(self.pos.x, self.pos.y, self.s, self.s)        
    
    def run(self):
        self.update()
        self.draw()

run()